src/web/app/pages-admin/admin-notifications-page/admin-notifications-page.component.spec.ts

```diff
@@ -1,30 +1,288 @@
+    jest.spyOn(notificationService, 'getNotifications').mockReturnValue(of({
+      notifications: [exampleNotificationOne, exampleNotificationTwo],
+    }));
+    component.initNotificationEditFormModel();
+    component.loadNotifications();
+    component.loadNotificationEditForm(component.notificationsTableRowModels[0].notification);
+    fixture.detectChanges();
+    expect(component.isNotificationEditFormExpanded).toBeTruthy();
+
+    const expectedMessage: string = component.notificationsTableRowModels[0].notification.message;
+
+    component.loadNotificationEditFormHandler(component.notificationsTableRowModels[0].notification);
+    component.notificationEditFormModel.message = 'new message';
+    expect(component.notificationEditFormModel.message).toEqual('new message');
+
+    // attempts to load edit form for another notification
+    component.loadNotificationEditFormHandler(component.notificationsTableRowModels[1].notification);
+    await promise;
+    expect(component.notificationsTableRowModels[0].notification.message).toEqual(expectedMessage);
+    expect(component.notificationEditFormModel.message)
+        .toEqual(component.notificationsTableRowModels[1].notification.message);
+
+    expect(modalSpy).toHaveBeenCalledTimes(1);
+    expect(modalSpy).toHaveBeenLastCalledWith('Discard unsaved edit?',
+        SimpleModalType.WARNING,
+        'Warning: If you choose to edit another notification, any unsaved changes will be lost.');
+  });
+
+  it('should delete notfication', () => {
+    component.notificationsTableRowModels = [notificationTableRowModel1];
+    expect(component.notificationsTableRowModels.length).toEqual(1);
+    jest.spyOn(notificationService, 'deleteNotification').mockReturnValue(of({
+      message: 'Successfully deleted',
+    }));
+    const spy: SpyInstance = jest.spyOn(statusMessageService, 'showSuccessToast')
+    .mockImplementation((args: string) => {
+      expect(args).toEqual('Successfully deleted');
+    });
+
+    component.deleteNotificationHandler(exampleNotificationEditModel.notificationId);
+    expect(spy).toHaveBeenCalledTimes(1);
+    expect(component.notificationsTableRowModels.length).toEqual(0);
+  });
+
+  it('should sort notification by start time', () => {
+    component.notificationsTableRowModels = [notificationTableRowModel1, notificationTableRowModel2];
+    component.notificationsTableRowModelsSortBy = SortBy.NOTIFICATION_CREATE_TIME;
+    component.sortNotificationsTableRowModelsHandler(SortBy.NOTIFICATION_START_TIME);
+    expect(component.notificationsTableRowModelsSortBy).toEqual(SortBy.NOTIFICATION_START_TIME);
+
+    const expected : NotificationsTableRowModel[] = [notificationTableRowModel1, notificationTableRowModel2]
+        .sort(component.getNotificationsTableRowModelsComparator());
+    expect(expected[0]).toBe(component.notificationsTableRowModels[0]);
+    expect(expected[1]).toBe(component.notificationsTableRowModels[1]);
```

Comment:
The effect of sorting is not obvious here? It seems that the original row models is `[notificationTableRowModel1, notificationTableRowModel2]` and after sorting it is still `[notificationTableRowModel1, notificationTableRowModel2]`, which didn't show any changes to show the effect of sorting.

Reply:
I see, I will reverse the order of two example notifications in `expected` to make it clearer